app/assets/javascripts/discourse/app/components/topic-drafts-dropdown.gjs

```diff
@@ -0,0 +1,108 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import DropdownMenu from "discourse/components/dropdown-menu";
+import DiscourseURL from "discourse/lib/url";
+import { i18n } from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+
+export default class TopicDraftsDropdown extends Component {
+  @service currentUser;
+  @service composer;
+
+  @tracked drafts = [];
+
+  get shouldDisplay() {
+    return this.args.showDraftsMenu;
+  }
+
+  get otherDraftsText() {
+    return this.args.otherDraftsCount > 0
+      ? i18n("drafts.dropdown.other_drafts", {
+          count: this.args.otherDraftsCount,
+        })
+      : "";
+  }
+
+  @action
+  onRegisterApi(api) {
+    this.dMenu = api;
+  }
+
+  @action
+  async onShowMenu() {
+    try {
+      const draftsStream = this.currentUser.get("userDraftsStream");
+      draftsStream.reset();
+
+      await draftsStream.findItems(this.site);
+      this.drafts = draftsStream.content.slice(0, this.args.draftLimit);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error("Failed to fetch drafts with error:", error);
+    }
+  }
+
+  @action
+  async resumeDraft(draft) {
+    await this.dMenu.close();
+
+    if (draft.get("postUrl")) {
```

Comment:
that's a detail but you don't need get for all these cases in action here.

An action will happen when you call it so you are not tracking changes.

Reply:
same for onShowMenu